### ISSUE TITLE
[Bug] Render process crashes in Edge (Chromium) on company self-hosted Atlassian Jira and Stash (Bitbucket)

### ISSUE THREAD

**Issue by NautillusSs** [darkreader/darkreader#10409]
### Prerequisites

- [X] I [searched for any existing report](https://github.com/darkreader/darkreader/issues?q=is%3Aissue) about this bug to avoid opening a duplicate.
- [X] I can reproduce this bug in a new, unmodified web browser profile with Dark Reader installed as the only extension.
- [X] I understand I need to use the [Broken Website Report template](https://github.com/darkreader/darkreader/issues/new?assignees=&labels=Broken+Website&template=broken-website-report.yml&title=%5BBroken+Website%5D+Replace+with+title) if this bug I am reporting occurs on a single website.

### Bug Description

I see 2 types of errors in the console:

# CORS (when fetching images to analyse them?)
Getting hundreds the lines below for images on the page after a while. Not sure why DarkReader needs to download the images again for analysis. Is it the tab sleep feature from Edge that forces the analysis to happen again?
I rewrote the original redirected URL as "original_URL_to_%2ficon--approved%2epng" for enhanced readability below.

```
Access to fetch at 'https://companyLoginHost.com/xxx?Parameters=...&original_URL_to_%2ficon--approved%2epng' (redirected from 'http://companyHostedJira.com/jira/xxx/icon_approved.png') from origin 'http://companyHostedJira.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
GET https://companyLoginHost.com/xxx?Parameters=...&original_URL_to_%2ficon--approved%2epng net::ERR_FAILED 302
```

# Uncaught TypeError: Cannot read properties of null (reading 'cssRules')

```
at VariablesStore.putRootVars
at createDynamicStyleOverrides
at runDynamicStyle
```

# Question (feature request)

Can I use the "export dynamic theme CSS" feature to export the CSS of above websites and use that instead of the dynamic theme? Can we somehow override the dynamic theme for specific websites and load our own exported CSS and use that as static theme?

Thank you.

### Website Address

Company self-hosted Atlassian Jira and Stash (Bitbucket)

### Steps To Reproduce

Not reproducible outside my company network I guess.

### Expected Behavior

Jira and Stash pages do not crash after a while.

### Actual Behavior

Jira and Stash pages crash after a while ("not enough memory").

### Operating System

Windows 10

### Web Browser name and version

Microsoft Edge 107.0.1418.42

### Dark Reader version

4.9.60

**Comment by Gusted** [comment]
> Getting hundreds the lines below for images on the page after a while. Not sure why DarkReader needs to download the images again for analysis.

There's caching on the background, but if it's able to download on the "frontground", it will do so -- there's no caching.

But this happens if assets are behind a authorization wall, which Dark Reader wouldn't be able to access as it explicitly tells the browser to not send any authorization(for safety reasons) .

> Getting a bunch of those errors in the console as well.

Are you able to get the full stacktrace? This seems like a bug.

> Can I use the "export dynamic theme CSS" feature to export the CSS of above websites and use that instead of the dynamic theme? Can we somehow override the dynamic theme for specific websites and load our own exported CSS and use that as static theme?

Yeah you can export it and use that instead, but you would need to use another addon like [Stylus](https://github.com/openstyles/stylus) to actually load the style into the site.

**Comment by NautillusSs** [comment]
Apologies for the delay in answering.
>Are you able to get the full stacktrace? This seems like a bug.

No sorry I haven't seen the issue again.
I'm seeing another issue where the dynamic theme does not trigger anymore after a while and the "fallback theme" does not disappear. I am talking about the following style node in the <head> of the page:
`<style class="darkreader darkreader--fallback" media="screen">...</style>`

At some point the dynamic theme stops working and that fallback style applies to all websites. It renders most websites colourless. On some pages it makes video players black (and cannot watch the video) or just makes the entire page black.

Was there a big change in the dynamic theme logic at some point? Our previous version didn't have those problems. Unfortunately I can't provide you with the exact version number anymore but it was obtained in 2019...

Thank you for your time.
